mongomock/filtering.py

```diff
@@ -287,6 +288,16 @@ def bson_compare(op, a, b, can_compare_types=True):
     if isinstance(a, NoneType):
         return op(0, 0)
 
+    # bson handles bytes as binary in python3+:
+    # https://api.mongodb.com/python/current/api/bson/index.html
+    if PY3 and isinstance(a, bytes):
+        # Performs the same operation as described by:
+        # https://docs.mongodb.com/manual/reference/bson-type-comparison-order/#bindata
+        if len(a) != len(b):
+            return op(len(a), len(b))
+        # bytes is always treated as subtype 0 by the bson library
```

Comment:
Drop those two lines, as the default is to run it in this function (see line 301).

Reply:
nope if  i remove these, the ordering differs with the mongo one (with a slight change to the tests that I am going to push).

Reply:
with current tests , i get (without these two lines):

```
======================================================================
FAIL: test__find_sort_list_empty_order (tests.test__mongomock.MongoClientCollectionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/mongomock/tests/test__mongomock.py", line 829, in test__find_sort_list_empty_order
    self.cmp.compare.find(sort=[('b', 1)])
  File "/data/mongomock/tests/multicollection.py", line 70, in __call__
    _assert_no_diff(results, ignore_order=self.___ignore_order, sort_by=self.___sort_by)
  File "/data/mongomock/tests/multicollection.py", line 95, in _assert_no_diff
    assert not d, _format_diff_message(prev_name, name, d)
AssertionError: Unexpected Diff:
	fake.3.b != real.3.b (b'alongbin1' != b'bin1')
	fake.4.b != real.4.b (b'alongbin2' != b'bin2')
	fake.5.b != real.5.b (b'bin1' != b'alongbin1')
	fake.6.b != real.6.b (b'bin2' != b'alongbin2')
```

Reply:
My point is that if you remove `'return op(a, b)`, Python will leave the `if PY3 and isinstance(a, bytes)` branch and go to the last line of the function which is also `return op(a, b)`, so I don't see how it goes wrong.

Reply:
Basically:

```py
if a:
  if b:
     return 0
  return 1
return 1
```

is equivalent to 

```py
if a:
  if b:
     return 0
return 1
```

Reply:
Ha ok. Dunno if it is due to an update, but the quoted code included the two lines before (and not the return) so i though you were talking about the 
```
if len(a) != len(b):
   return op(len(a), len(b))
```
lines

Sorry, should have guessed it. Will do !